core/arch/arm/kernel/boot.c

```diff
@@ -656,11 +656,13 @@ static int init_dt_overlay(struct dt_descriptor *dt, int __maybe_unused dt_size)
 	int fragment;
 	int ret;
 
-	ret = fdt_check_header(dt->blob);
-	if (!ret) {
-		fdt_for_each_subnode(fragment, dt->blob, 0)
-			dt->frag_id += 1;
-		return ret;
+	if (!IS_ENABLED(CFG_GENERATE_DTB_OVERLAY)) {
+		ret = fdt_check_header(dt->blob);
+		if (!ret) {
+			fdt_for_each_subnode(fragment, dt->blob, 0)
+				dt->frag_id += 1;
+			return ret;
+		}
 	}
 
 #ifdef CFG_DT_ADDR
```

Comment:
I don't think it really makes sense the secure boot stage passes a DTB address in r2/x2 when the area is not filled with a DTB. But maybe the OP-TEE implementation can allow that.

Reply:
Actually, that would make sense to avoid hardcoding the external device tree location with CFG_DT_ADDR.

Reply:
There is some lelgacy in imx for 2 paltforms enabling `CFG_EXTERNAL_DTB_OVERLAY` and setting `CFG_DT_ADDR`.
Do you always expect to use boot argument arg#2 (arm register r2/x2) to pass DTB location to use with `CFG_GENERATE_DTB_OVERLAY=y`?

Reply:
What I meant is to support both (either specified in r2 or CFG_DT_ADDR). Currently, only the `ifdef` forbid using this feature without CFG_DT_ADDR. Removing the ifdef would allow both r2/CFG_DT_ADDR to work with CFG_EXTERNAL_DTB_OVERLAY/CFG_GENERATE_DTB_OVERLAY I guess. Do you see any potential problem here ?